extensions/wikia/CreateNewWiki/CreateNewWikiController.class.php

```diff
@@ -120,21 +120,60 @@ private function setupVerticalsAndCategories() {
 		$verticalToCategoriesSetMapping = array( 2 => 1, 7 => 1, 4 => 1, 3 => 1, 1 => 1, 6 => 1, 5 => 2, 0 => 2 );
 
-		$this->verticals = array();
+		/**
+		 * Current keys for translating Vertical ID to string:
+		 * 'oasis-label-wiki-vertical-id-0' => 'Other',
+		 * 'oasis-label-wiki-vertical-id-1' => 'TV',
+		 * 'oasis-label-wiki-vertical-id-2' => 'Video Games',
+		 * 'oasis-label-wiki-vertical-id-3' => 'Books',
+		 * 'oasis-label-wiki-vertical-id-4' => 'Comics',
+		 * 'oasis-label-wiki-vertical-id-5' => 'Lifestyle',
+		 * 'oasis-label-wiki-vertical-id-6' => 'Music',
+		 * 'oasis-label-wiki-vertical-id-7' => 'Movies',
+		 */
+		$this->verticals = [];
 		foreach($verticalsOrder as $verticalId) {
-			$this->verticals[] = array(
-				'id' => $allVerticals[$verticalId]['id'],
-				'name' => $allVerticals[$verticalId]['name'],
-				'short' => $allVerticals[$verticalId]['short'],
+			$verticalData = $allVerticals[$verticalId];
+			$this->verticals[] = [
+				'id' => $verticalData['id'],
+				'name' => wfMessage( 'oasis-label-wiki-vertical-id-' . $verticalData['id'] )->text(),
+				'short' => $verticalData['short'],
 				'categoriesSet' => $verticalToCategoriesSetMapping[$verticalId]
-			);
+			];
 		}
 
-		$this->categoriesSets = array();
+		/**
+		 * Current keys for translating Category ID to string:
+		 * 'oasis-label-wiki-category-id-1' => 'Humor',
+		 * 'oasis-label-wiki-category-id-5' => 'Toys',
+		 * 'oasis-label-wiki-category-id-6' => 'Food and Drink',
+		 * 'oasis-label-wiki-category-id-7' => 'Travel',
+		 * 'oasis-label-wiki-category-id-8' => 'Education',
+		 * 'oasis-label-wiki-category-id-10' => 'Finance',
+		 * 'oasis-label-wiki-category-id-11' => 'Politics',
+		 * 'oasis-label-wiki-category-id-12' => 'Technology',
+		 * 'oasis-label-wiki-category-id-13' => 'Science',
+		 * 'oasis-label-wiki-category-id-14' => 'Philosophy',
+		 * 'oasis-label-wiki-category-id-15' => 'Sports',
+		 * 'oasis-label-wiki-category-id-16' => 'Music',
+		 * 'oasis-label-wiki-category-id-17' => 'Creative',
+		 * 'oasis-label-wiki-category-id-18' => 'Auto',
+		 * 'oasis-label-wiki-category-id-21' => 'TV',
+		 * 'oasis-label-wiki-category-id-22' => 'Video Games',
+		 * 'oasis-label-wiki-category-id-23' => 'Books',
+		 * 'oasis-label-wiki-category-id-24' => 'Comics',
+		 * 'oasis-label-wiki-category-id-25' => 'Fanon',
+		 * 'oasis-label-wiki-category-id-26' => 'Home and Garden',
+		 * 'oasis-label-wiki-category-id-27' => 'Movies',
+		 * 'oasis-label-wiki-category-id-28' => 'Anime',
+		 */
+		$this->categoriesSets = [];
 		foreach($categoriesSetsOrder as $setId => $categoriesOrder) {
-			$categoriesSet = array();
-			foreach($categoriesOrder as $categoryId) {
-				$categoriesSet[] = $allCategories[$categoryId];
+			$categoriesSet = [];
+			foreach($categoriesOrder as $categoryId => $categoryData) {
+				$categoryData = $allCategories[$categoryId];
+				$categoryData['name'] = wfMessage( 'oasis-label-category-id-' . $categoryData['id'] )->text();
```

Comment:
https://one.wikia-inc.com/wiki/Engineering/Onboarding/Internationalization/Technical/Advanced#Outputting_messages_to_HTML - I'm not sure if escaped() in necessary but it's going to be placed in HTML